### DISCUSSION TITLE
What makes Bun so much faster?

### DISCUSSION THREAD

**Discussion by jacoscaz** [comunica/comunica#1261] [General]
Now that [Bun has hit 1.x](https://bun.sh/blog/bun-v1.0), I thought it would be a good idea to benchmark Quadstore + Comunica on the two runtimes.

One of [quadstore's performance tests](https://github.com/jacoscaz/quadstore-perf) looks at how fast Quadstore + Comunica are able to run through the entire set of quads in the store with a simple `SELECT * WHERE { ?s ?p ?o . }` query. I had to use the in-memory LevelDB adapter, though, as Bun still crashes when using LevelDB proper.

To my surprise, when run on Bun, Quadstore + Comunica consistently score ~4.5x faster than when run on Node.js. Quadstore itself, without Comunica, is able to run through all quads almost 10x faster on Bun than on Node.js.

I'm not sure what to do with this but it felt worth highlighting. Perhaps there are tricks we can learn by looking at how Bun / WebKit do things internally?

**Comment by rubensworks** [comment]
That's a very significant difference!

I wonder what the performance difference would be when running more complex queries, such as the ones provided by [WatDiv](https://dsg.uwaterloo.ca/watdiv/).
